### ISSUE TITLE
Confusing output about promotion when a subproject fails to configure

### ISSUE THREAD

**Issue by jon-turney** [mesonbuild/meson#3095]
```console
# git clone https://github.com/valum-framework/valum
Cloning into 'valum'...
remote: Counting objects: 10216, done.
remote: Compressing objects: 100% (25/25), done.
remote: Total 10216 (delta 8), reused 12 (delta 5), pack-reused 10186
Receiving objects: 100% (10216/10216), 3.34 MiB | 2.22 MiB/s, done.
Resolving deltas: 100% (7262/7262), done.
# cd valum
# meson _build
The Meson build system
Version: 0.45.0.dev1
Source dir: /valum
Build dir: /valum/_build
Build type: native build
Project name: valum
Native C compiler: cc (gcc 7.2.0 "cc (Ubuntu 7.2.0-8ubuntu3) 7.2.0")
Native Vala compiler: valac (valac 0.36.6)
Build machine cpu family: x86
Build machine cpu: i686
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency glib-2.0 found: YES 2.54.1
Native dependency gobject-2.0 found: YES 2.54.1
Native dependency gio-2.0 found: YES 2.54.1
Native dependency libsoup-2.4 found: YES 2.60.1
Cloning into 'vsgi'...
remote: Counting objects: 5749, done.
remote: Total 5749 (delta 0), reused 0 (delta 0), pack-reused 5749
Receiving objects: 100% (5749/5749), 1.76 MiB | 1.86 MiB/s, done.
Resolving deltas: 100% (3896/3896), done.
Already on 'master'
Your branch is up-to-date with 'origin/master'.

Executing subproject vsgi.

Project name: vsgi
Native C compiler: cc (gcc 7.2.0 "cc (Ubuntu 7.2.0-8ubuntu3) 7.2.0")
Native Vala compiler: valac (valac 0.36.6)
Dependency glib-2.0 found: YES (cached)
Dependency gobject-2.0 found: YES (cached)
Dependency gio-2.0 found: YES (cached)
Native dependency gio-unix-2.0 found: YES 2.54.1
Native dependency gmodule-2.0 found: YES 2.54.1
Dependency libsoup-2.4 found: YES (cached)
Dependency libsystemd found: NO
Also couldn't find a fallback subproject in subprojects/vsgi for the dependency vsgi
Reason: Native dependency 'openssl' not found                                         <----

Dependency vsgi not found but it is available in a sub-subproject.
To use it in the current project, promote it by going in the project source
root and issuing one of the following commands.

meson wrap promote subprojects/vsgi
meson wrap promote subprojects/vsgi.wrap

meson.build:17:0: ERROR: Native dependency 'vsgi' not found
```

After PR #2334 it is assumed that a subproject has failed to configure because it's not a top-level subproject.

If it failed for any other reason (e.g. a missing dependency) then `print_nested_info()` shouldn't be called, as this obscures the actual error, and offers incorrect advice about remediation.

**Comment by jon-turney** [comment]
With PR #3213 the output for the same build attempts is:

```
The Meson build system
Version: 0.46.0.dev1
Source dir: /valum
Build dir: /valum/_build
Build type: native build
Project name: valum
Native C compiler: cc (gcc 7.2.0 "cc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0")
Native Vala compiler: valac (valac 0.36.6)
Build machine cpu family: x86
Build machine cpu: i686
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency glib-2.0 found: YES 2.54.1
Native dependency gobject-2.0 found: YES 2.54.1
Native dependency gio-2.0 found: YES 2.54.1
Native dependency libsoup-2.4 found: YES 2.60.1
Cloning into 'vsgi'...
remote: Counting objects: 5749, done.
remote: Total 5749 (delta 0), reused 0 (delta 0), pack-reused 5749
Receiving objects: 100% (5749/5749), 1.76 MiB | 943.00 KiB/s, done.
Resolving deltas: 100% (3896/3896), done.
Already on 'master'
Your branch is up-to-date with 'origin/master'.

Executing subproject vsgi.

Project name: vsgi
Native C compiler: cc (gcc 7.2.0 "cc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0")
Native Vala compiler: valac (valac 0.36.6)
Dependency glib-2.0 found: YES (cached)
Dependency gobject-2.0 found: YES (cached)
Dependency gio-2.0 found: YES (cached)
Native dependency gio-unix-2.0 found: YES 2.54.1
Native dependency gmodule-2.0 found: YES 2.54.1
Dependency libsoup-2.4 found: YES (cached)
Dependency libsystemd found: NO
Also couldn't find a fallback subproject in subprojects/vsgi for the dependency vsgi 
Reason: Native dependency 'openssl' not found

meson.build:17:0: ERROR: Native dependency 'vsgi' not found
``` 

This is better, but still confusing in a few ways:
* we don't mention that we are trying the subproject until after cloning the subproject repo
* subproject build output could perhaps be indented/prefixed to make it more distinct
* failure in a subproject is reported in a different format to failure in the top-level project
* that format doesn't give source location, which I think is just going to be inconvenient if there's an actual bug in the subproject's meson.build
* "couldn't find a fallback" doesn't make sense in this context. The subproject was found it, but couldn't be configured


 format

**Comment by jon-turney** [comment]
There a still a couple of items left in the ticklist.

I started looking at using `mlog.exception()` to report the subproject failure, but that unfortunately run into a problem with VS2017 I found intractable.

See https://github.com/jon-turney/meson/commit/0026ba497e4c982fe69c3e49d8bd9b5f27a4d443

**Comment by nirbheek** [comment]
Apparently this is how you work around it: https://github.com/Microsoft/msbuild/issues/2424#issuecomment-322226483

**Comment by jon-turney** [comment]
Yeah, I came across that.  It doesn't seem to work, I think perhaps because we are using `CustomBuild` tasks rather than `Exec` tasks, but I'd hit my 'I don't care about VS' limit at that point...

**Comment by nirbheek** [comment]
Maybe we can strip the "ERROR:" prefix and only print the message? Printing "ERROR" when it's not an error is probably confusing anyway?

**Comment by jon-turney** [comment]
> I started looking at using `mlog.exception()` to report the subproject failure

Fixed as of 815f1205a8d2eb78fa71057c593f4eec3746aad3 (#4593)
